File: SWEA/D3/2806. N－Queen/N－Queen.py
def check(x):
  for i in range(x):
    if d[x] == d[i] or abs(d[x]-d[i]) == x-i:
      return False
  return True
def dfs(x):
  global cnt
  if x == n:
    cnt += 1
  else:
    for i in range(n):
      d[x] = i
      if check(x):
        dfs(x+1)
for t in range(int(input())):
  n = int(input())
  d = [0]*n
  cnt = 0
  dfs(0)
  print(f'#{t+1} {cnt}')

File: SWEA/D3/2806. N－Queen/N－Queen.cpp
#include <iostream>
#include <algorithm>
using namespace std;
int t,n,cnt = 0;
int d[11];
bool check(int x) {
    for (int i = 0; i < x; i++) {
        if (d[x] == d[i] || abs(d[x] - d[i]) == x - i)
            return false;
    }
    return true;
}
void dfs(int x) {
    if (x == n) {
        cnt++;
        return;
    }
    else {
        for (int i = 0; i < n; i++) {
            d[x] = i;
            if (check(x))
                dfs(x + 1);
        }
    }

}
int main() {
    cin >> t;
    for (int i = 0; i < t; i++) {
        cin >> n;
        cnt = 0;
        dfs(0);
        cout << "#" << i + 1 << " " << cnt << '\n';
    }
    return 0;
}